### ISSUE TITLE
ParsingConnection doesn't correctly relay Statement::getResultSet, Statement::getUpdateCount, or Statement::getMoreResults on static Statement

### ISSUE THREAD

**Issue by theodiefenthal** [jOOQ/jOOQ#13974]
### Expected behavior

No exception :) 

### Actual behavior

I have a liquibase file with only one changeSet with only one SQL statement written excplicitly in it.  That SQL statement is written in MariaDB style syntax. (CREATE TABLE). 

To query the external MariaDB database a customer provided to me, I thought of creating JOOQ classes for it. I hence put the SQL statemement in Liquibase and wanted to use JOOQs parsing connection to transform the MariaDB style create table statement in native H2 syntax into a H2 database from which I would then let run JOOQ schema generation (as in project build time, I have no access tu (customer) MariaDB). 

Against a local (real mariadb), I created the connection for my liquibase updater in a MVP like so:
```
    protected Connection createLiquibaseConnection() throws SQLException {
        try {
            Class.forName("org.mariadb.jdbc.Driver");
        } catch (ClassNotFoundException e) {
            throw new IllegalStateException("for " + getClass() + " to work properly, you have to manually include the database dependency (e.g. h2 or mariadb driver)", e);
        }
        return DriverManager.getConnection(jdbcConnection);
    }
```

I now changed that to something like this:
```
            protected Connection createLiquibaseConnection() throws SQLException {
                try {
                    Class.forName("org.h2.Driver");
                } catch (ClassNotFoundException e) {
                    throw new IllegalStateException("for " + getClass() + " to work properly, you have to manually include the database dependency (e.g. h2 or mariadb driver)", e);
                }

                return DSL.using(jdbcConnection).parsingConnection();
            }
```
where my `jdbcConnection` (String) is: `jdbc:h2:file:....;INIT=CREATE SCHEMA IF NOT EXISTS schemagen;SET SCHEMA schemagen`

If I now start the application, Liquibase won't even execute my single one statement. Instead, I receive the following output:
```
16:33:08.552 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:08.552 [main] DEBUG org.jooq.impl.ParsingConnection - Translation cache miss   : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:08.623 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:08.682 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : CREATE TABLE SCHEMAGEN.DATABASECHANGELOGLOCK (ID INT NOT NULL, LOCKED BOOLEAN NOT NULL, LOCKGRANTED TIMESTAMP, LOCKEDBY VARCHAR(255), CONSTRAINT PK_DATABASECHANGELOGLOCK PRIMARY KEY (ID))
16:33:08.682 [main] DEBUG org.jooq.impl.ParsingConnection - Translation cache miss   : CREATE TABLE SCHEMAGEN.DATABASECHANGELOGLOCK (ID INT NOT NULL, LOCKED BOOLEAN NOT NULL, LOCKGRANTED TIMESTAMP, LOCKEDBY VARCHAR(255), CONSTRAINT PK_DATABASECHANGELOGLOCK PRIMARY KEY (ID))
16:33:08.708 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : create table SCHEMAGEN.DATABASECHANGELOGLOCK (ID int not null, LOCKED boolean not null, LOCKGRANTED timestamp, LOCKEDBY varchar(255), constraint PK_DATABASECHANGELOGLOCK primary key (ID))
16:33:09.076 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.076 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.081 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.081 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.084 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : DELETE FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.085 [main] DEBUG org.jooq.impl.ParsingConnection - Translation cache miss   : DELETE FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.120 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : delete from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.334 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.334 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.335 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.335 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.336 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : DELETE FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.336 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : delete from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.662 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.662 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.663 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.663 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.664 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : DELETE FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.664 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : delete from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.671 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.671 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.671 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.672 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.672 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : DELETE FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:09.672 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : delete from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:10.669 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:10.670 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:10.671 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:10.672 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:10.674 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : DELETE FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:10.674 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : delete from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:11.309 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:11.309 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:11.310 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:11.311 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:11.313 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : DELETE FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:11.313 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : delete from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:11.486 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:11.486 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:11.488 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:11.488 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:11.490 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : DELETE FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:11.490 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : delete from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:12.401 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:12.402 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:12.403 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:12.404 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:12.406 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : DELETE FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:12.407 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : delete from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:12.922 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:12.923 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:12.924 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : SELECT COUNT(*) FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:12.924 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : select count(*) from SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:12.926 [main] DEBUG org.jooq.impl.ParsingConnection - Translating from         : DELETE FROM SCHEMAGEN.DATABASECHANGELOGLOCK
16:33:12.926 [main] DEBUG org.jooq.impl.ParsingConnection - Translating to           : delete from SCHEMAGEN.DATABASECHANGELOGLOCK
Exception in thread "main" liquibase.exception.LockException: liquibase.exception.DatabaseException: No PreparedStatement is available yet [Failed SQL: (0) DELETE FROM SCHEMAGEN.DATABASECHANGELOGLOCK]
	at liquibase.lockservice.StandardLockService.acquireLock(StandardLockService.java:329)
	at liquibase.lockservice.StandardLockService.waitForLock(StandardLockService.java:247)
	at liquibase.Liquibase.lambda$update$1(Liquibase.java:212)
	at liquibase.Scope.lambda$child$0(Scope.java:180)
	at liquibase.Scope.child(Scope.java:189)
	at liquibase.Scope.child(Scope.java:179)
	at liquibase.Scope.child(Scope.java:158)
	at liquibase.Liquibase.runInScope(Liquibase.java:2414)
	at liquibase.Liquibase.update(Liquibase.java:209)
	at liquibase.Liquibase.update(Liquibase.java:195)
	at org.example.mvp.JooqBindingGenerator.run(JooqBindingGenerator.java:115)
	at org.example.MainJooqWithH2andJOOQparsingConnection.main(MainJooqWithH2andJOOQparsingConnection.java:65)
Caused by: liquibase.exception.DatabaseException: No PreparedStatement is available yet [Failed SQL: (0) DELETE FROM SCHEMAGEN.DATABASECHANGELOGLOCK]
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:440)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:78)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:161)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:126)
	at liquibase.lockservice.StandardLockService.init(StandardLockService.java:132)
	at liquibase.lockservice.StandardLockService.acquireLock(StandardLockService.java:286)
	... 11 more
Caused by: java.sql.SQLException: No PreparedStatement is available yet
	at org.jooq.impl.ParsingStatement.last(ParsingStatement.java:351)
	at org.jooq.impl.ParsingStatement.getUpdateCount(ParsingStatement.java:486)
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:437)
	... 16 more
```

### Steps to reproduce the problem

Use a parsing connection feature from for executing liquibase changesets. 

With the created connection, I did the following (Liquibase 4.15.0):
```
        liquibase.database.Database database = DatabaseFactory.getInstance().findCorrectDatabaseImplementation(new JdbcConnection(connection));

        String physicalFilePath = changeset.toString();
        logger.info("running {}", physicalFilePath);

        Liquibase liquibase = new Liquibase(physicalFilePath, new FileSystemResourceAccessor(currentWorkingDirectory.toFile()), database);
            liquibase.update(new Contexts("!test"), new LabelExpression());  // <--- Here it crashes
```

### jOOQ Version

JOOQ Professional 3.17.4

### Database product and version

H2 2.1.214

### Java Version

OpenJDK 11

### OS Version

Windows 10 x64

### JDBC driver name and version (include name if unofficial driver)

H2 2.1.214

**Comment by theodiefenthal** [comment]
If you can't reproduce, let me know. I will then investigate some more time reducing the MVP to something I can upload here.

**Comment by lukaseder** [comment]
I can reproduce this by using JDBC API only, like this:

```java
public void testTranslateSQLExecuteJDBCDirectlyStaticStatementExecuteAndUpdateCount() throws Exception {
    try (Connection c = create(settings()).parsingConnection()) {
        try (Statement s = c.createStatement()) {
            assertTrue(s.execute("SELECT first_name || ' ' || last_name FROM t_author WHERE id < 3 ORDER BY id ASC"));
            try (ResultSet rs = s.getResultSet()) {
                Result<Record> r = create().fetch(rs);
                assertEquals(
                    asList("George Orwell", "Paulo Coelho"),
                    r.getValues(0));

                assertFalse(s.getMoreResults());
                assertEquals(-1, s.getUpdateCount());
            }

    // ...
```

Key here is the usage of static statements instead of `PreparedStatement`, and also `execute()` instead of `executeQuery()` or `executeUpdate()`.

There was a test doing the same thing with a `PreparedStatement`, which didn't fail. This should obviously be fixed ASAP, because that JDBC usage might be quite popular among third parties, like Liquibase, Flyway, and others.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.5 (https://github.com/jOOQ/jOOQ/issues/13983), and 3.16.11 (https://github.com/jOOQ/jOOQ/issues/13984)
